JavaScript/JavaScriptSDK/Context/User.ts

```diff
@@ -51,6 +51,12 @@ module Microsoft.ApplicationInsights.Context {
                 date.setTime(date.getTime() + 31536000000);
                 var newCookie = [this.id, acqStr];
                 Util.setCookie('ai_user', newCookie.join('|') + ';expires=' + date.toUTCString());
+
+                // If we have an ai_session in local storage this means the user actively removed our cookies.
+                // We should respect their wishes and clear ourselves from local storage
+                if (window.localStorage && localStorage['ai_session']) {
+                    localStorage.removeItem('ai_session');
```

Comment:
I think we want to abstract localStorage usage in separate function. Also as we discussed I think we should add try/catch around accessing localStorage to catch unexpected IO errors and let execution flow proceed